app/models/trainee.rb

```diff
@@ -10,6 +10,7 @@ class Trainee < ApplicationRecord
   has_many :nationalities, through: :nationalisations
   has_many :trainee_disabilities, dependent: :destroy, inverse_of: :trainee
   has_many :disabilities, through: :trainee_disabilities
+  belongs_to :lead_school, optional: true, class_name: "School"
```

Comment:
I'm still not sure whether this is the best design. Doing it like this means that this field (and `employing_school` if we follow the pattern) will be null for potentially thousands of trainee records. Did you try using a join table and compare the two approaches? (we already do that for disabilities, nationalities, subjects etc)

Reply:
Discussed and voted offline. Consensus is to go with foreign key